### PR TITLE
Cleanup encodings init()

### PR DESCRIPTION
**Pull request by b4n**

### CODE REVIEW

**Comment by codebrainz** [comment]
Not tested, but LGTM at a glance.

File: src/encodings.c [hunk 1]
```diff
@@ -428,85 +434,61 @@ void encodings_init(void)
...
+				GtkWidget *item = gtk_menu_item_new_with_mnemonic(_(groups[i]));
+				submenus[i] = gtk_menu_new();
+				gtk_menu_item_set_submenu(GTK_MENU_ITEM(item), submenus[i]);
+				gtk_container_add(GTK_CONTAINER(menu[k]), item);
+				gtk_widget_show_all(item);
 			}
+		}
 
-			while (order < group_size)	/* the biggest group has 13 elements */
+		/** TODO can it be optimized? ATM 882 runs at line "if (encoding->group ...)" */
```
Comment (elextr) [inline src/encodings.c]: Is this comment still relevant, and if so which "if" does it refer to?  AFAICT there is no `if(encoding->group` line in the following loop.

Comment (b4n) [inline src/encodings.c]: well, it's still not great so maybe.  And yeah, fixed the reference that I changed at some point but changed it back later.

**Comment by b4n** [comment]
@codebrainz fair enough, I used shiny C99 for loop declarations all through this function :)

**Pull request merged**